Mathlib/CategoryTheory/FiberedCategory/HomLift.lean

```diff
@@ -0,0 +1,233 @@
+  obtain rfl : R = S := by rw [← codomain_eq p (𝟙 R) φ, domain_eq p f ψ]
+  simpa using inferInstanceAs (p.IsHomLift (𝟙 R ≫ f) (φ ≫ ψ))
+
+lemma eqToHom_domain_lift_id {p : 𝒳 ⥤ 𝒮} {a b : 𝒳} (hab : a = b) {R : 𝒮} (hR : p.obj a = R) :
+    p.IsHomLift (𝟙 R) (eqToHom hab) where
+  fac := by simp [eqToHom_map]
+
+lemma eqToHom_codomain_lift_id {p : 𝒳 ⥤ 𝒮} {a b : 𝒳} (hab : a = b) {S : 𝒮} (hS : p.obj b = S) :
+    p.IsHomLift (𝟙 S) (eqToHom hab) where
+  fac := by simp [eqToHom_map]
+
+lemma id_lift_eqToHom_domain {p : 𝒳 ⥤ 𝒮} {R S : 𝒮} (hRS : R = S) {a : 𝒳} (ha : p.obj a = R) :
+    p.IsHomLift (eqToHom hRS) (𝟙 a) where
+  fac := by simp [eqToHom_map]
+
+lemma id_lift_eqToHom_codomain {p : 𝒳 ⥤ 𝒮} {R S : 𝒮} (hRS : R = S) {b : 𝒳} (hb : p.obj b = S) :
+    p.IsHomLift (eqToHom hRS) (𝟙 b) where
+  fac := by simp [eqToHom_map]
+
+instance comp_eqToHom_lift {R S : 𝒮} {a' a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : a' = a)
+    [p.IsHomLift f φ] : p.IsHomLift f (eqToHom h ≫ φ) :=
+  have := eqToHom_codomain_lift_id h (domain_eq p f φ)
+  id_comp f ▸ comp (eqToHom h) φ
+
+instance eqToHom_comp_lift {R S : 𝒮} {a b b' : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : b = b')
+    [p.IsHomLift f φ] : p.IsHomLift f (φ ≫ eqToHom h) :=
+  have := eqToHom_domain_lift_id h (codomain_eq p f φ)
+  comp_id f ▸ comp φ (eqToHom h)
+
+instance lift_eqToHom_comp {R' R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : R' = R)
+    [p.IsHomLift f φ] : p.IsHomLift ((eqToHom h) ≫ f) φ :=
+  have := id_lift_eqToHom_codomain h (domain_eq p f φ)
+  id_comp φ ▸ comp (𝟙 a) φ
+
+instance lift_comp_eqToHom {R S S': 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : S = S')
+    [p.IsHomLift f φ] : p.IsHomLift (f ≫ (eqToHom h)) φ :=
+  have := id_lift_eqToHom_domain h (codomain_eq p f φ)
+  comp_id φ ▸ comp φ (𝟙 b)
+
+@[simp]
+lemma comp_eqToHom_lift_iff {R S : 𝒮} {a' a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : a' = a) :
+    p.IsHomLift f (eqToHom h ≫ φ) ↔ p.IsHomLift f φ where
+  mp := by intro hφ'; subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma eqToHom_comp_lift_iff {R S : 𝒮} {a b b' : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : b = b') :
+    p.IsHomLift f (φ ≫ eqToHom h) ↔ p.IsHomLift f φ where
+  mp := by intro hφ'; subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma lift_eqToHom_comp_iff {R' R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : R' = R) :
+    p.IsHomLift ((eqToHom h) ≫ f) φ ↔ p.IsHomLift f φ where
+  mp := by intro hφ'; subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+@[simp]
+lemma lift_comp_eqToHom_iff {R S S' : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ⟶ b) (h : S = S') :
+    p.IsHomLift (f ≫ (eqToHom h)) φ ↔ p.IsHomLift f φ where
+  mp := by intro hφ'; subst h; simpa using hφ'
+  mpr := fun hφ => inferInstance
+
+/-- The isomorphism `R ≅ S` obtained from an isomorphism `φ : a ≅ b` lifting `f` -/
+def isoOfIsoLift  {R S : 𝒮} {a b : 𝒳} (f : R ⟶ S) (φ : a ≅ b) [p.IsHomLift f φ.hom] :
```

Comment:
I would suggest you define direclty both the `hom` and `inv` fields of this `isoOfIsoLift`, so that the proof of `isoOfIsoLift_hom` becomes `rfl`. This may facilitate some future proofs, because `dsimp` would already be able to simplify.